src/bluetooth.cpp

```diff
@@ -172,16 +155,34 @@ void bluetooth::finished() {
         forceHeartBeltOffForTimeout = true;
     }
 
-#ifndef Q_OS_IOS
-    if (!trx_route_key && !bh_spada_2 && !technogym_myrun_treadmill_experimental) {
-#endif
-        discoveryAgent->start(QBluetoothDeviceDiscoveryAgent::LowEnergyMethod);
-#ifndef Q_OS_IOS
-    } else {
-        discoveryAgent->start(QBluetoothDeviceDiscoveryAgent::ClassicMethod |
-                              QBluetoothDeviceDiscoveryAgent::LowEnergyMethod);
-    }
-#endif
+    this->startDiscovery();
+}
+
+void bluetooth::startDiscovery() {
+
+    #ifndef Q_OS_IOS
+        QSettings settings;
+        bool technogym_myrun_treadmill_experimental =
+            settings.value(QStringLiteral("technogym_myrun_treadmill_experimental"), false).toBool();
+        bool trx_route_key = settings.value(QStringLiteral("trx_route_key"), false).toBool();
+        bool bh_spada_2 = settings.value(QStringLiteral("bh_spada_2"), false).toBool();
+
+        if (!trx_route_key && !bh_spada_2 && !technogym_myrun_treadmill_experimental) {
+    #endif
+            discoveryAgent->start(QBluetoothDeviceDiscoveryAgent::LowEnergyMethod);
+    #ifndef Q_OS_IOS
+        } else {
+            discoveryAgent->start(QBluetoothDeviceDiscoveryAgent::ClassicMethod |
+                                  QBluetoothDeviceDiscoveryAgent::LowEnergyMethod);
+        }
+    #endif
+}
+
+
+void bluetooth::stopDiscovery()
+{
+    if(this->discoveryAgent)
+        this->discoveryAgent->stop();
```

Comment:
This is changed behaviour...previous code was not doing the nullptr check.

Reply:
let's put a qdebug if the discoveryagent is a null pointer in order to track future bug, thanks

Reply:
Would you prefer one that always happens, or all call to debug(..) which checks the log field ?

Reply:
In the else case as you did, thanks!